accepted/future-releases/abstract-external-fields/feature-specification.md

```diff
@@ -0,0 +1,240 @@
+implement the accessors. Instead, `D` tacity inherits the implementation of
+the concrete variable. This may be a bug, because `C` is designed for having
+these accessors overridden to do specific things that the concrete variable
+will not do. Also, if `D` _does_ override the accessors with a getter and
+setter as intended, the storage reserved for the concrete instance variable
+will be a space leak (it is unused).
+
+Finally, if null-safety is enabled then a concrete instance variable
+declaration (say, `int foo;`) is a compile-time error if the type of the
+variable is non-nullable, unless the variable is initialized in the
+initializer list of each generative constructor of the class. This problem
+arises both in connection with `dart:ffi` and for a an instance variable
+which is used as a replacement for an abstract getter and possibly setter.
+
+So the existing workaround of using a concrete instance variable to emulate
+an abstract instance variable is inconvenient, error prone, and confusing
+for a reader of the code, and for a writer of a subclass, and similar
+problems exist when a regular variable is used as a replacement for an
+external getter and possibly setter.
+
+
+## Design Idea
+
+In response to the issues above, this document introduces abstract and
+external variables.
+
+The basic idea is that an abstract variable is syntactic sugar for an
+abstract getter and possibly an abstract setter, and similarly for an
+external variable, in both cases such that there is no storage and no
+implementation of the getter and possible setter, and they have the same
+signatures as the ones that would be induced implicitly by a concrete
+variable declaration.
+
+The next section is the normative text that specifies the syntax and
+semantics that realize this idea.
+
+
+## Feature Specification
+
+An _abstract instance variable declaration_ is an instance variable
+declaration prefixed by the modifier `abstract`. It must not be late, and
+it cannot have an initializer expression.
+
+An _external variable declaration_ is a non-local, non-parameter variable
+declaration prefixed by the modifier `external`. It must not be abstract,
+const, or late, and it cannot have an initializer expression.
+
+The syntax and behavior of these constructs is specified in the following
+sections.
+
+
+### Syntax
+
+The grammar is modified as follows:
+
+```
+<topLevelDefinition> ::=
+  ... |
+  // New alternative.
+  'external' <finalVarOrType> <identifierList> ';'
+
+<finalVarOrType> ::= // New rule.
+  'final' <type>? |
+  <varOrType>
+
+<declaration> ::=
+  ... |
+  // New alternative.
+  'external' ('static'? <finalVarOrType> | 'covariant' <varOrType>)
+      <identifierList> |
+  // New alternative.
+  'abstract' (<finalVarOrType> | 'covariant' <varOrType>) <identifierList>
+```
+
+
+### Static Analysis
+
+The features specified in this document are syntactic sugar, that is, they
+are specified in terms of a small program transformation that eliminates
```

Comment:
eliminates them -> reduces them to a different syntactic construct.

(there's still something left afterwards, so it's not *eliminiated*.)

Reply:
Done in new PR.